lib/spack/spack/test/llnl/util/lock.py

```diff
@@ -1345,8 +1345,7 @@ def _lockf(fd, cmd, len, start, whence):
     with tmpdir.as_cwd():
         lockfile = "lockfile"
         lock = lk.Lock(lockfile)
-
-        touch(lockfile)
+        lock.acquire_read()
 
         monkeypatch.setattr(fcntl, "lockf", _lockf)
```

Comment:
:grimacing:

Reply:
Otherwise we fail on the assertions. Without assertions `mypy` gets mad :slightly_smiling_face:

Reply:
Yeah, I mean the issue is that we shouldn't be monkeypatching python internals, that's a big red flag.